de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoice/model/validator/C_Invoice.java

```diff
@@ -254,4 +257,29 @@ private void deleteInvoiceLines(final I_C_Invoice invoice)
 			InterfaceWrapperHelper.delete(line);
 		}
 	}
+
+	@DocValidate(timings = { ModelValidator.TIMING_BEFORE_COMPLETE })
+	public void linkInvoiceToPaymentIfNeeded(final I_C_Invoice invoice)
+	{
+		final I_C_Order order = invoice.getC_Order();
```

Comment:
* what if there are >orders and therefore `invoice.getC_Order_ID() <= 0`? 
* what if there are >1 invoices for one order and therefore only the order's paid amount needs to be distributed among them?

Talked with @metas-mk :
> what if there are >orders and therefore `invoice.getC_Order_ID() <= 0`?
* for a prepay order there may be just one invoice. so we need to make sure that even if the customer has "Sammelrechnung erlaubt", the system still needs to use the "no-aggregation" `C_Aggegration` for the prepay order's ICs

> what if there are >1 invoices for one order and therefore only the order's paid amount needs to be distributed among them?

* AFAIR long time ago we created `C_AllocationLine.C_Order_ID` for the prepay order use case.. i think it wasn't a great idea; can you please take a look and remove `C_AllocationLine.C_Order_ID` if it's still there?